#Michelin Guide Restaurants - Data Analysis and Visualization

**Kaggle Dataset**

Jerry Ng. (2025). Michelin Guide Restaurants [Data set]. Kaggle. https://doi.org/10.34740/KAGGLE/DSV/10347142

In [ ]:
import pandas as pd

df = pd.read_csv('michelin_my_maps.csv', encoding='utf-8')

column_names = df.columns.tolist()
print(column_names)

unique_awards = df['Award'].unique()

print(unique_awards)

['Name', 'Address', 'Location', 'Price', 'Cuisine', 'Longitude', 'Latitude', 'PhoneNumber', 'Url', 'WebsiteUrl', 'Award', 'GreenStar', 'FacilitiesAndServices', 'Description']
['3 Stars' '2 Stars' '1 Star' 'Bib Gourmand' 'Selected Restaurants']


**Not all columns in the dataset are relevant to the analysis and visualizations, so the irrelevant columns have been removed.**

**Also, the Awards column contained rows with 1 Star, 2 Stars, 3 Stars, Selected Restaurants, and Bib Gourmand. The restaurants in these categories have been removed. The analysis concentrated on restaurants with 1, 2, or 3 stars, not other Michelin categories.**

**The column Price contained different currency symbols, depending on the country of the restaurants — 1 to 4 symbols. The symbols have been replaced with Inexpensive, Moderately Expensive, Expensive, and Very Expensive to facilitate the analysis when comparing restaurants from countries with different currencies.**

In [ ]:
import warnings
warnings.filterwarnings("ignore", category=FutureWarning)

import pandas as pd

df = pd.read_csv('michelin_my_maps.csv', encoding='utf-8')

columns_to_remove = ['Address', 'PhoneNumber', 'FacilitiesAndServices', 'Description', 'WebsiteUrl', 'GreenStar', 'Url']
df = df.drop(columns=columns_to_remove, errors='ignore')

df = df[~df['Award'].str.contains('Selected Restaurants|Bib Gourmand')]

award_mapping = {'1 Star': 1, '2 Stars': 2, '3 Stars': 3}
df['Award'] = df['Award'].replace(award_mapping)

price_mapping = {
    '$': 'Inexpensive',
    '$$': 'Moderately Expensive',
    '$$$': 'Expensive',
    '$$$$': 'Very Expensive',
    '฿': 'Inexpensive',
    '฿฿': 'Moderately Expensive',
    '฿฿฿': 'Expensive',
    '฿฿฿฿': 'Very Expensive',
    '€': 'Inexpensive',
    '€€': 'Moderately Expensive',
    '€€€': 'Expensive',
    '€€€€': 'Very Expensive',
    '¥': 'Inexpensive',
    '¥¥': 'Moderately Expensive',
    '¥¥¥': 'Expensive',
    '¥¥¥¥': 'Very Expensive',
    '£': 'Inexpensive',
    '££': 'Moderately Expensive',
    '£££': 'Expensive',
    '££££': 'Very Expensive',
    '₺': 'Inexpensive',
    '₺₺': 'Moderately Expensive',
    '₺₺₺': 'Expensive',
    '₺₺₺₺': 'Very Expensive',
    '﷼': 'Inexpensive',
    '﷼﷼': 'Moderately Expensive',
    '﷼﷼﷼': 'Expensive',
    '﷼﷼﷼﷼': 'Very Expensive',
    '₩': 'Inexpensive',
    '₩₩': 'Moderately Expensive',
    '₩₩₩': 'Expensive',
    '₩₩₩₩': 'Very Expensive',
    '₫': 'Inexpensive',
    '₫₫': 'Moderately Expensive',
    '₫₫₫': 'Expensive',
    '₫₫₫₫': 'Very Expensive'
}

df['Price'] = df['Price'].replace(price_mapping)

print (df)

                                  Name                Location  \
0                     Jungsik New York           New York, USA   
1                               Per Se           New York, USA   
2                         Le Bernardin           New York, USA   
3                  Eleven Madison Park           New York, USA   
4                                 Masa           New York, USA   
...                                ...                     ...   
3592                              TIAN         Vienna, Austria   
3593                      Kilian Stuba      Hirschegg, Austria   
3594                     Pfefferschiff       Salzburg, Austria   
3595                             Field  Prague, Czech Republic   
3596  La Degustation Bohême Bourgeoise         Prague, Czechia   

                     Price                     Cuisine  Longitude   Latitude  \
0           Very Expensive        Korean, Contemporary -74.009111  40.718726   
1           Very Expensive        Contemporary,

**The visualizations have been created with Plotly.**

In [ ]:
!pip install plotly==5.15.0
import plotly.express as px
import pandas as pd

In [ ]:
import warnings
warnings.filterwarnings("ignore", category=FutureWarning)

import pandas as pd
import plotly.express as px

df = pd.read_csv('michelin_my_maps.csv', encoding='utf-8')

columns_to_remove = ['Address', 'PhoneNumber', 'FacilitiesAndServices', 'Description', 'WebsiteUrl', 'GreenStar']
df = df.drop(columns=columns_to_remove, errors='ignore')

df = df[~df['Award'].str.contains('Selected Restaurants|Bib Gourmand')]

award_mapping = {'1 Star': 1, '2 Stars': 2, '3 Stars': 3}
df['Award'] = df['Award'].replace(award_mapping)

price_mapping = {
    '$': 'Inexpensive',
    '$$': 'Moderately Expensive',
    '$$$': 'Expensive',
    '$$$$': 'Very Expensive',
    '฿': 'Inexpensive',
    '฿฿': 'Moderately Expensive',
    '฿฿฿': 'Expensive',
    '฿฿฿฿': 'Very Expensive',
    '€': 'Inexpensive',
    '€€': 'Moderately Expensive',
    '€€€': 'Expensive',
    '€€€€': 'Very Expensive',
    '¥': 'Inexpensive',
    '¥¥': 'Moderately Expensive',
    '¥¥¥': 'Expensive',
    '¥¥¥¥': 'Very Expensive',
    '£': 'Inexpensive',
    '££': 'Moderately Expensive',
    '£££': 'Expensive',
    '££££': 'Very Expensive',
    '₺': 'Inexpensive',
    '₺₺': 'Moderately Expensive',
    '₺₺₺': 'Expensive',
    '₺₺₺₺': 'Very Expensive',
    '﷼': 'Inexpensive',
    '﷼﷼': 'Moderately Expensive',
    '﷼﷼﷼': 'Expensive',
    '﷼﷼﷼﷼': 'Very Expensive',
    '₩': 'Inexpensive',
    '₩₩': 'Moderately Expensive',
    '₩₩₩': 'Expensive',
    '₩₩₩₩': 'Very Expensive',
    '₫': 'Inexpensive',
    '₫₫': 'Moderately Expensive',
    '₫₫₫': 'Expensive',
    '₫₫₫₫': 'Very Expensive'
}

df['Price'] = df['Price'].replace(price_mapping)

fig = px.scatter_mapbox(
    df,
    lat="Latitude",
    lon="Longitude",
    color="Price",
    hover_name="Name",
    hover_data={'Location': True, 'Price': True, 'Cuisine': True, 'Award': True,
                'Latitude': False, 'Longitude': False},
    mapbox_style="open-street-map",
    zoom=10
)

fig.update_layout(title="Restaurant Locations and Prices")
fig.show()

fig.write_html("Location_and_Prices.html")


**The location of the restaurants was based on latitude and longitude, and the map shows the restaurant locations on the world map. The colors are blue for very expensive, red for expensive, green for moderately expensive, and purple for inexpensive.**

**When hovering over, one can see the name of the restaurant, location, cuisine, and award (1, 2, or 3 stars).**

**The visualization shows that most of the restaurants with Michelin stars are located in Europe.**

In [ ]:
import plotly.express as px
import pandas as pd

price_counts = df['Price'].value_counts()

fig = px.bar(x=price_counts.index, y=price_counts.values,
             labels={'x': 'Price Range', 'y': 'Number of Restaurants'},
             title='Number of Restaurants by Price Range',
             category_orders={"x": ["Very Expensive", "Expensive", "Moderately Expensive", "Inexpensive"]})

fig.write_html("NumberofRestaurantsbyPriceRange.html")

fig.show()

**Most of the restaurants with Michelin stars are categorized as very expensive (2,299), followed by expensive (1,198), moderately expensive (98), and only 2 are categorized as inexpensive.**

In [ ]:
award_counts = df['Award'].value_counts()

star_ratings = award_counts.index.astype(str)

fig = px.bar(x=award_counts.index, y=award_counts.values,
             labels={'x': 'Award Level', 'y': 'Number of Restaurants'},
             title='Number of Restaurants by Award Level')

fig.update_traces(text=star_ratings, textposition='auto')
fig.update_layout(legend_title_text='Award')

fig.show()
fig.write_html("NumberofRestaurantsbyAwardLevel.html")

**On the other hand, the data shows how hard it must be to earn 3 Michelin stars — 148 restaurants have 3 stars, followed by 488 restaurants with 2 stars, and 2,961 with 1 star.**

In [ ]:
import plotly.express as px

cuisine_counts = df['Cuisine'].value_counts()

fig = px.pie(
    values=cuisine_counts.values,
    names=cuisine_counts.index,
    title='Number of Restaurants by Cuisine Type'
)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(uniformtext_minsize=12, uniformtext_mode='hide')

fig.show()
fig.write_html("NumberofRestaurantsbyCuisineType.html")

In [ ]:
num_cuisine_categories = df['Cuisine'].nunique()
print(f"There are {num_cuisine_categories} different cuisine categories.")

There are 537 different cuisine categories.


**The types of cuisine in restaurants with Michelin stars are vast (537). The three most frequent types are Modern Cuisine (17.9%), Creative (10.2%), and Japanese (6.26%).**

The analysis highlights key insights into Michelin-starred restaurants worldwide. Most of these restaurants are located in Europe, reflecting the region's prominence in fine dining. A significant portion of Michelin-starred restaurants are categorized as very expensive, emphasizing the premium nature of these establishments.

Additionally, earning a Michelin star is exceptionally challenging, with only 148 restaurants achieving the coveted 3-star status, 488 earning 2 stars, and 2,961 earning 1 star. The diversity of cuisines offered is remarkable, with Modern Cuisine, Creative, and Japanese being the most frequent types among Michelin-starred restaurants.

This analysis provides valuable perspectives for understanding the global distribution, pricing, and culinary variety of Michelin-starred establishments, offering a foundation for further exploration of trends and patterns in the fine dining industry.